### PR TITLE
drivers: sensor: adxl367: initial implementation

### PR DESCRIPTION
**Pull request by amiclaus**
The ADXL367 is an ultralow power, 3-axis microelectromechanical systems (MEMS) accelerometer that consumes only 0.89 μA at
a 100 Hz output data rate and 180 nA when in motion-triggered wake-up mode. Unlike accelerometers that use power duty cycling to achieve low power consumption, the ADXL367 does not alias input signals by undersampling, but samples the full bandwidth of the sensor at all data rates.

### CODE REVIEW

File: drivers/sensor/adxl367/adxl367.c [hunk 1]
```diff
@@ -0,0 +1,1000 @@
...
+
+
+	ret = data->hw_tf->write_reg_mask(dev, ADXL367_ACT_INACT_CTL,
+					  ADXL367_ACT_INACT_CTL_ACT_EN_MSK |
+					  ADXL367_ACT_INACT_CTL_ACT_REF_MSK,
+					  FIELD_PREP(ADXL367_ACT_INACT_CTL_ACT_EN_MSK, th->enable) |
+					  FIELD_PREP(ADXL367_ACT_INACT_CTL_ACT_REF_MSK,
+						     th->referenced));
+	if (ret)
+		return ret;
```
Comment (MaureenHelm) [inline drivers/sensor/adxl367/adxl367.c:50]: Need braces even for single line blocks. See [Coding Style](https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style)

File: drivers/sensor/adxl367/adxl367.c [hunk 2]
```diff
@@ -0,0 +1,1000 @@
...
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		adxl367_accel_convert(dev, val, data->sample.z);
+		break;
+	case SENSOR_CHAN_ACCEL_XYZ:
+		adxl367_accel_convert(dev, val++, data->sample.x);
+		adxl367_accel_convert(dev, val++, data->sample.y);
+		adxl367_accel_convert(dev, val, data->sample.z);
+		break;
+	case SENSOR_CHAN_AMBIENT_TEMP:
```
Comment (MaureenHelm) [inline drivers/sensor/adxl367/adxl367.c:801]: `SENSOR_CHAN_DIE_TEMP`?

File: drivers/sensor/adxl367/adxl367_i2c.c [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+};
+
+int adxl367_i2c_init(const struct device *dev)
+{
+	struct adxl367_data *data = dev->data;
+	const struct adxl367_dev_config *config = dev->config;
+
+	data->hw_tf = &adxl367_i2c_transfer_fn;
+
+	if (!device_is_ready(config->i2c.bus))
```
Comment (MaureenHelm) [inline drivers/sensor/adxl367/adxl367_i2c.c:96]: `i2c_is_ready_dt`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please add this sensor to the build test to ensure it gets covered in CI:
- `tests/drivers/build_all/sensor/i2c.dtsi`
- `tests/drivers/build_all/sensor/spi.dtsi`

**Review by amiclaus — COMMENTED** [review]
changes in v2:
- use braces as per coding style.
- switch to `SENSOR_CHAN_DIE_TEMP` attribute
- use `i2c_is_ready_dt`
- add i2c/spi test nodes

**Review by amiclaus — COMMENTED** [review]
changes in v3:
- add extra test cs-gpio for adxl372 spi node in app.overlay
- add missing brace
- fix i2c test node.

**Review by MaureenHelm — DISMISSED** [review]

**Review by amiclaus — COMMENTED** [review]
changes in v4:
- fix newly appeared conflicts in `tests/drivers/build_all/sensor/i2c.dtsi`

**Review by MaureenHelm — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Pull request merged**